### PR TITLE
[Merged by Bors] - chore: fix nonterminal simps

### PR DESCRIPTION
**Pull request by BoltonBailey**
Fixes the nonterminal simps identified by #7496 

---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by BoltonBailey** [comment]
Output:
```
Error: Mathlib/Algebra/Category/ModuleCat/FilteredColimits.lean#L104: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/Category/ModuleCat/Limits.lean#L1[15](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:16): ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/Hom/Centroid.lean#L398: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/Homology/Single.lean#L440: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/MonoidAlgebra/Basic.lean#L603: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/MonoidAlgebra/Basic.lean#L805: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/MonoidAlgebra/Basic.lean#L1120: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/MonoidAlgebra/Basic.lean#L1912: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/Order/Floor.lean#L307: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Algebra/Order/LatticeGroup.lean#L279: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/AlgebraicTopology/CechNerve.lean#L172: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Analysis/NormedSpace/AddTorsor.lean#L301: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Analysis/NormedSpace/Banach.lean#L113: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Analysis/NormedSpace/Multilinear.lean#L218: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Analysis/NormedSpace/Multilinear.lean#L1484: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Analysis/NormedSpace/Multilinear.lean#L1487: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Analysis/SpecificLimits/Basic.lean#L272: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Equivalence.lean#L261: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/FintypeCat.lean#L[16](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:17)1: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/FintypeCat.lean#L168: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Grothendieck.lean#L198: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Grothendieck.lean#L208: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Grothendieck.lean#L213: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Localization/Construction.lean#L[17](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:18)8: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Monoidal/Free/Basic.lean#L348: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Monoidal/Limits.lean#L84: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Monoidal/Limits.lean#L94: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Monoidal/Mon_.lean#L383: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Monoidal/Mon_.lean#L452: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/CategoryTheory/Sites/Adjunction.lean#L132: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Halting.lean#L[18](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:19)3: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Halting.lean#L256: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Partrec.lean#L227: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/PartrecCode.lean#L1128: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/PartrecCode.lean#L1137: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Primrec.lean#L346: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Primrec.lean#L357: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Primrec.lean#L571: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/Primrec.lean#L914: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1359: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1369: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1395: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1417: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1427: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1447: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1486: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1503: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1512: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1529: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1531: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1551: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1561: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1570: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1589: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1644: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1651: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1737: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TMToPartrec.lean#L1743: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Computability/TuringMachine.lean#L1843: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Bitvec/Lemmas.lean#L46: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Fin/Basic.lean#L1660: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Finset/Basic.lean#L2805: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Finset/Basic.lean#L2933: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Finset/NoncommProd.lean#L430: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Fintype/Powerset.lean#L66: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/List/Chain.lean#L355: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/List/NatAntidiagonal.lean#L89: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Multiset/Pi.lean#L127: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Multiset/Sections.lean#L47: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Nat/ModEq.lean#L461: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/PEquiv.lean#L395: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/PFunctor/Multivariate/M.lean#L338: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Data/Seq/Parallel.lean#L144: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Init/Data/Nat/Bitwise.lean#L132: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Init/Data/Nat/Bitwise.lean#L311: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/LinearAlgebra/Orientation.lean#L[20](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:21)4: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/LinearAlgebra/QuadraticForm/Basic.lean#L579: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/LinearAlgebra/TensorAlgebra/Basic.lean#L209: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Logic/Encodable/Lattice.lean#L45: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Logic/Equiv/Option.lean#L115: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Logic/Equiv/TransferInstance.lean#L537: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Logic/Equiv/TransferInstance.lean#L579: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/MeasureTheory/Measure/Hausdorff.lean#L1096: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/MeasureTheory/Measure/OuterMeasure.lean#L298: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/MeasureTheory/Measure/OuterMeasure.lean#L533: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/MeasureTheory/Measure/OuterMeasure.lean#L685: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/MeasureTheory/Measure/OuterMeasure.lean#L1245: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/MeasureTheory/Measure/OuterMeasure.lean#L1602: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/ModelTheory/DirectLimit.lean#L298: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/ModelTheory/DirectLimit.lean#L349: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/NumberTheory/Dioph.lean#L451: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/NumberTheory/Dioph.lean#L548: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/NumberTheory/Dioph.lean#L575: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/NumberTheory/Zsqrtd/Basic.lean#L1043: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Order/LocallyFinite.lean#L1450: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Order/LocallyFinite.lean#L1457: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Order/LocallyFinite.lean#L1464: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Order/LocallyFinite.lean#L1471: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Order/PFilter.lean#L177: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/SetTheory/Ordinal/Basic.lean#L650: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/SetTheory/Ordinal/Basic.lean#L1268: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/SetTheory/Ordinal/Notation.lean#L7[22](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:23): ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Topology/Algebra/Module/Basic.lean#L2596: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Topology/Algebra/ValuedField.lean#L290: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Mathlib/Topology/Sheaves/SheafCondition/EqualizerProducts.lean#L334: ERR_NSP: Non-terminal simp. Replace with `simp?` and use the suggested output
Error: Process completed with exit code 1[23](https://github.com/leanprover-community/mathlib4/actions/runs/6400378822/job/17373904318?pr=7496#step:7:24).
```

File: Mathlib/Algebra/Order/LatticeGroup.lean [hunk 1]
```diff
@@ -276,7 +276,7 @@ theorem mul_inf_eq_mul_inf_mul [CovariantClass α α (· * ·) (· ≤ ·)] (a b
     c * (a ⊓ b) = c * a ⊓ c * b := by
   refine' le_antisymm _ _
   rw [le_inf_iff, mul_le_mul_iff_left, mul_le_mul_iff_left]
-  simp
+  simp only [ge_iff_le, inf_le_left, inf_le_right, and_self]
```
Comment (ocfnash) [inline Mathlib/Algebra/Order/LatticeGroup.lean:279]: I note that `simp?` has a bug where it often redundantly includes `ge_iff_le`. Can it be removed here (and in two other spots below)?
```suggestion
  simp only [inf_le_left, inf_le_right, and_self]
```

**Review by ocfnash — APPROVED** [review]
Thanks!

bors d+

File: Mathlib/Algebra/Order/LatticeGroup.lean (hunk 1 above)
Comment (BoltonBailey) [reply Mathlib/Algebra/Order/LatticeGroup.lean:279]: Done!

**Pull request closed without merging**